cognite/client/utils/_client_config.py

```diff
@@ -116,7 +116,7 @@ def __init__(
             )
 
             if token_generator.token_params_set():
-                self.token = lambda: token_generator.return_access_token()
+                self.token = token_generator.return_access_token
```

Comment:
this won't work because the issue is not just about lambda. you [cannot pickle any nested function](https://stackoverflow.com/a/12022055/15604332), which is what we have in return_access_token.

Reply:
it'd be better to make changes to multiprocessing code in such a way that:
1. it first gets the token
2. then spawns up new processes that use that token.
@eighty20results

Reply:
I seem to be a able to pickle it after the change:
```py
$ python aaaaaaaa.py
hey hoo from 'if token_generator.token_params_set()' WITH LAMBDA
Trying to pickle...
Traceback (most recent call last):
  File "aaaaaaaa.py", line 25, in <module>
    pickle.dumps(client)
AttributeError: Can't pickle local object 'ClientConfig.__init__.<locals>.<lambda>'
$ 
$ python aaaaaaaa.py
hey hoo from 'if token_generator.token_params_set()' WITHOUT LAMBDA
Trying to pickle...
SUCCESS!!!
```

Reply:
okay, let's try it then. hope it works fine.